src/Toml/Parser/Value.hs

```diff
@@ -22,13 +22,31 @@ import Control.Applicative.Combinators (between, count, option, optional, sepBy1
 import Data.Fixed (Pico)
 import Data.Time (Day, LocalTime (..), TimeOfDay, ZonedTime (..), fromGregorianValid,
                   makeTimeOfDayValid, minutesToTimeZone)
+import Data.Char (ord)
+
 import Text.Read (readMaybe)
 
-import Toml.Parser.Core (Parser, binary, char, digitChar, hexadecimal, lexeme, octal, sc, signed,
+import Toml.Parser.Core (Parser, char, digitChar, hexDigitChar, octDigitChar, binDigitChar, lexeme, sc, signed,
                          string, text, try, (<?>))
 import Toml.Parser.String (textP)
 import Toml.Type (AnyValue, UValue (..), typeCheck)
 
+-- | Returns decimal value from base N representation
+baseStringToInt :: Integer -> String -> Maybe Integer
+baseStringToInt base = Just . getValue . reverse
```

Comment:
Now I implemented only one function for hex, oct and bin numbers but without ``foldl'`` because I used ``megaparsec`` parser to convert in from ``[Char]`` or ``String`` into ``Integer``.